app/assets/javascripts/discourse/widgets/post.js.es6

```diff
@@ -219,6 +220,68 @@ function showReplyTab(attrs, siteSettings) {
   );
 }
 
+createWidget("post-date", {
+  tagName: "div.post-info.post-date",
+
+  buildClasses(attrs) {
+    let classes = "post-date";
+
+    const lastWikiEdit =
+      attrs.wiki && attrs.lastWikiEdit && new Date(attrs.lastWikiEdit);
+
+    if (lastWikiEdit) {
+      classes = `${classes} last-wiki-edit`;
+    }
+
+    return classes;
+  },
+
+  html(attrs) {
+    return h("a", {}, dateNode(this._date(attrs)));
+  },
+
+  _date(attrs) {
+    const lastWikiEdit =
+      attrs.wiki && attrs.lastWikiEdit && new Date(attrs.lastWikiEdit);
+    const createdAt = new Date(attrs.created_at);
+    return lastWikiEdit ? lastWikiEdit : createdAt;
+  },
+
+  click() {
+    const post = this.findAncestorModel();
+
+    const modalFallback = () => {
+      showModal("share-and-invite", {
+        modalClass: "share-and-invite",
+        panels: [
+          {
+            id: "share",
+            title: "topic.share.extended_title",
+            model: {
+              postNumber: this.attrs.post_number,
+              shareUrl: this.attrs.shareUrl,
+              date: this._date(this.attrs),
+              postId: post.get("id"),
+              topic: post.get("topic")
+            }
+          }
+        ]
+      });
+    };
+
+    // use native webshare when available
+    // navigator.share needs HTTPS, returns undefined on HTTP
+    if (window.navigator.share) {
```

Comment:
This might be better as a function that you could re-use in the couple of places you are doing the same check.

```js
import navigatorShare from ...

navigatorShare(this.attrs.shareUrl).catch(() => modalFallback());
```